### ISSUE TITLE
API Key?!

### ISSUE THREAD

**Issue by EdwardOak** [Safeer-Abbas/Kijij-All-In-One#1]
Hey! I've been dreaming about getting a kijiji reposter up and running. 

I'm using George Lim's kijiji-bot and I get the recaptcha error also. 

I'm checking with your all-in-one script but I can't find anything about a Kijiji API key, I can't even find mentions of API on kijiji website. 

What am I missing?

**Comment by Safeer-Abbas** [comment]
Hey I actually took a break on this project. Currently expanding my knowledge with how API works, etc. Ill fix george lims code and send it to you in a few seconds. You will need a 2captcha api key inorder to have the captchas solved, etc

**Comment by Safeer-Abbas** [comment]
https://2captcha.com/ 

Its a captcha solving platform. You use there api key.

**Comment by Safeer-Abbas** [comment]
When you have a 2captcha key let me know, ill compile both of our codes into one for you. I will also start working on this making it into a fully finished AIO bot

**Comment by EdwardOak** [comment]
Amazing, I'm getting a key in a hot minute.

What then? Thanks a lot, my friend and I will benefit from this.

**Comment by Safeer-Abbas** [comment]
ight bet. Once you have it let me know. Do NOT paste the key here. You should not give the key to anyone

**Comment by EdwardOak** [comment]
I got the key.

What next?

In the meantime I'll try to find a better way to bypass this very low-level recaptcha issue without a paid (real person) proxy if possible

⁣Sent from BlueMail ​

**Comment by Safeer-Abbas** [comment]
I'll look into sniffing out the mobile API, that would not have captcha on it.

Add the following line of code into the payload of the login.

`"recaptchaResponse": captcha_solver(YOUR API KEY, "6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "https://www.kijiji.ca/t-login.html"),

So now it should look like this
https://prnt.sc/vqlhpl

Now replace YOUR API KEY with a string that contains your 2captcha api key so now you should have 

`"recaptchaResponse": captcha_solver("yourapikerey....", "6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "https://www.kijiji.ca/t-login.html"),

Finally add my Captcha file in this kiji repository to the same folder as his code and add this line of import

from Captcha import captcha_solver.

Now his code should work 100%

**Comment by EdwardOak** [comment]
Alright I'm doing that now.

In which file is the login's payload?
__init__.py
__main__.py
apicalls.py

**Comment by EdwardOak** [comment]
Confirm this is the accurate captcha file you're talking about:
https://github.com/Safeer-Abbas/Kijij-All-In-One/blob/master/Kiji%20All%20In%20One%20Manager/Captcha.py

**Comment by Safeer-Abbas** [comment]
Yes thats the correct file. The payload is in gerogelim bot, go into the kiji_bot folder, then kijiapi.py and then scroll down until you see payload.

https://github.com/george-lim/kijiji-bot/blob/master/kijiji_bot/kijiji_api.py

**Comment by EdwardOak** [comment]
Got it, I think.


class KijijiApi:
    """
    All functions require to be logged in to Kijiji first in order to
function correctly
    """
    def __init__(self):
        config = {}
        self.session = requests.Session()

    def login(self, username, password):
        """
        Login to Kijiji for the current session
        """
        login_url = 'https://www.kijiji.ca/'
        resp = self.session.get(login_url, headers=request_headers)
        payload = {
            "operationName": "loginUser",
            "variables": {
                "input": {
                    "emailOrNickname": username,
                    "password": password,
                    "rememberMe": True,
                    "targetUrl": None,
                    "fraudToken": None,  # Valid value doesn't appear to be
necessary for login
                    "campaign": None,
                    "xsrfToken": get_xsrf_token(resp.text)
                }
            },
   "query": "mutation loginUser($input: LoginUserInput!) {\n
 loginUser(input: $input) {\n    userId\n    message\n    statusCode\n
 redirectUrl\n    __typename\n  }\n}\n",
        }
        api_url = 'https://www.kijiji.ca/anvil/api'  # API endpoint
        resp = self.session.post(api_url, json=payload)

        if not self.is_logged_in():
            raise KijijiApiException("Could not log in.", resp.text)


Where exactly do I add the:
`"recaptchaResponse": captcha_solver(MY API KEY,
"6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT",
"https://www.kijiji.ca/t-login.html"),
?

**Comment by Safeer-Abbas** [comment]
below remember me above tagert url

**Comment by Safeer-Abbas** [comment]
like that
![image](https://user-images.githubusercontent.com/58538577/100384058-39e51080-2fed-11eb-8258-e802d7c390f5.png)

**Comment by EdwardOak** [comment]
Ok, 1 min.

**Comment by EdwardOak** [comment]
import captcha_solver (in kijiji-api.py OR __main__.py?)

captcha.py file (no edits? correct name? specific location?)

**Comment by Safeer-Abbas** [comment]
from Captcha import captcha_solver.

^ Put that in the first line of the kiji_bot of georgelims. Make sure the the name is Captcha.py and its in the same location as georgelims bot

**Comment by EdwardOak** [comment]
Just confirming, you mean __main__.py (in /exemples/repostads/...)?

**Comment by Safeer-Abbas** [comment]
Paste it here

https://github.com/george-lim/kijiji-bot/blob/master/kijiji_bot/kijiji_api.py

**Comment by EdwardOak** [comment]
*This:*
"recaptchaResponse": captcha_solver(f3459f465aa90662743ac478088cf872,
"6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "
https://www.kijiji.ca/t-login.html",
"targetUrl": None,

*Throws this error when runnin __main__.py:*
  File "/usr/local/lib/python3.6/dist-packages/kijiji_bot/kijiji_api.py",
line 109
    "targetUrl": None,
               ^
SyntaxError: invalid syntax

**Comment by Safeer-Abbas** [comment]
Your missing a bracket

**Comment by EdwardOak** [comment]
edwardoak/bot/kijiji-api.py

I'm getting a syntax error on line 109 (targeturl in payload)

**Comment by Safeer-Abbas** [comment]
Its because you are missing a bracket after the kiji link, you opened one but never closed it

**Comment by EdwardOak** [comment]
again, apologies, I'm loonie.

**Comment by EdwardOak** [comment]
1 sec

**Comment by Safeer-Abbas** [comment]
Dont worry, george-lims bot is a bit hard to understand and read. I will definetly work on the bot today and make it very user-friendly.

**Comment by EdwardOak** [comment]
"recaptchaResponse": captcha_solver(f3459f465aa90662743ac478088cf872),
"6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "
https://www.kijiji.ca/t-login.html",


                                                               ^
SyntaxError: invalid syntax

**Comment by EdwardOak** [comment]
I added bracket but still get syntax error (?)

**Comment by EdwardOak** [comment]
I'll switch the API key (i just realized I didnt censor it)

**Comment by Safeer-Abbas** [comment]
The bracket has to be after the .html" part, so it will be 

`"recaptchaResponse": captcha_solver("f3459f465aa90662743ac478088cf872", "6LeqLCwUAAAAADHwg3Gj-TciS00CC1MiHhn3wNT", "https://www.kijiji.ca/t-login.html")`

**Comment by EdwardOak** [comment]
sorry

**Comment by Safeer-Abbas** [comment]
Nah man dont worry. I was confused too once when I was dealing with python. Ill make sure the kiji bot I make is packed into an exe that way you just need to double click the file and its like a normal tool ( will have options to sign in, repost, etc)

**Comment by EdwardOak** [comment]
comma at the end also, right?

"recaptchaResponse": captcha_solver("f3459f465aa90662743ac478088cf872",
"6LeqLCwUAAAAADHwg3Gj-TciS00CC1MiHhn3wNT", "
https://www.kijiji.ca/t-login.html") *,*

**Comment by Safeer-Abbas** [comment]
Yes comma at the end

**Comment by EdwardOak** [comment]
I specialize in SMS/VoIP, marketing campaigns & carrier filters, etc...

If you need help for campaigns don't hesitate to contact me.

Question:
I get a "captcha_solver is not defined" from kijiji-api.py.

How do I define it?

I added

from Captcha.py import captcha_solver

but only in __main__.py

not kijiji-api.py

Do I add it in kijiji-api.py?

**Comment by Safeer-Abbas** [comment]
Yeah add it in kijiji-api.py. The main.py file gets all its information from kiji-api.py. Also make sure the Captcha.py file is in the same folder as kjij-api.py

Also it should be the following
`from Captcha import captcha_solver` #no .py at the end of Captcha

**Comment by EdwardOak** [comment]
File "/usr/local/lib/python3.6/dist-packages/kijiji_bot/kijiji_api.py",
line 109, in login
    "recaptchaResponse": captcha_solver(f3459f465aa90662743ac478088cf872,
"6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "
https://www.kijiji.ca/t-login.html"),
NameError: name 'f3459f465aa90662743ac478088cf872' is not defined

(That's where YOUR API KEY was // That's my key)

>

**Comment by EdwardOak** [comment]
Do I need to add it in Captcha.py where (API_KEY) is?

**Comment by EdwardOak** [comment]
def captcha_solver(API_KEY, site_key, page_url): >>> def captcha_solver(
f3459f465aa90662743ac478088cf872 , site_key, page_url):

?

**Comment by Safeer-Abbas** [comment]
No, dont touch the capthca.py file. Make sure your code is in a string ( so have the qutations around it)

**Comment by Safeer-Abbas** [comment]
Make sure your api key* has quotations around it

**Comment by EdwardOak** [comment]
Got it.

**Comment by EdwardOak** [comment]
File "/home/laplumeoak/kijiji-bot/examples/repost_ads/Captcha.py", line
5, in captcha_solver
    captcha_id = requests.get(f"
http://2captcha.com/in.php?key={API_KEY}&method=userrecaptcha&googlekey={site_key}&pageurl={page_url}
").text.split('|')[1]
IndexError: list index out of range

**Comment by EdwardOak** [comment]
sitekey issue?

**Comment by Safeer-Abbas** [comment]
Make sure you are using your newest api key

**Comment by Safeer-Abbas** [comment]
How urgently do you need this?

**Comment by EdwardOak** [comment]
I am using the only key I have.

This is my main project.

I'm not forcing you though, take your time, do your things.

I'll try to troubleshoot (i'll back everything up)

thanks a million my man

**Comment by Safeer-Abbas** [comment]
oh no I absolutley love helping you. I was just thinking if you can wait around 2-3 days. My entire project will be done. It will be very very user friendly and easy to use.

**Comment by Safeer-Abbas** [comment]
But what error are you getting right now?

**Comment by EdwardOak** [comment]
To be entirely honest, having it as pyscripts is great for
troubleshooting ++ you help me understand how it works (by allowing me
attention).

Now I get:

 File "/home/laplumeoak/kijiji-bot/examples/repost_ads/Captcha.py", line 5,
in captcha_solver
    captcha_id = requests.get(f"
http://2captcha.com/in.php?key={API_KEY}&method=userrecaptcha&googlekey={site_key}&pageurl={page_url}
").text.split('|')[1]
*IndexError: list index out of range  *

**Comment by EdwardOak** [comment]
But if you finish before me, I'll take it also, lol

**Comment by Safeer-Abbas** [comment]
The entire project should be done in roughly 2-3 days. It will be in python and the src code will be available too. In the meantime you can use this. Regarding the error, make sure you have the latest 2captcha api key

**Comment by EdwardOak** [comment]
[image: image.png]

"recaptchaResponse": captcha_solver("f3459f465aa90662743ac478088cf872",
"6LeqLCwUAAAAADHwg3Gj-DTciS00CC1MiHhn3wNT", "
https://www.kijiji.ca/t-login.html"),

**Comment by EdwardOak** [comment]
The error is thrown from Captcha.py using "name(?)" : myapikey

**Comment by Safeer-Abbas** [comment]
you need to change the 2captcha api key again. Make sure you copy the latest key from 2captcha and paste it in the payload of recaptchaResponse

**Comment by EdwardOak** [comment]
Ok.

Do I need 2captcha credits or is a few allowed for free?

**Comment by Safeer-Abbas** [comment]
You will need to load up your 2captcha balance up. Its super cheap too around 1000 recpatchas solved for 50 cents.

**Comment by EdwardOak** [comment]
Same error with new key.

**Comment by EdwardOak** [comment]
ok 1sec

**Comment by Safeer-Abbas** [comment]
make sure it has some balance on it

**Comment by EdwardOak** [comment]
Hey! 

Sorry I was busy. 

I just added credits to 2captcha and wish me luck!

I'm doing a test right now.

Did you compile an exe or nah?

⁣Sent from BlueMail ​

**Comment by EdwardOak** [comment]
Dump >>

{"errors":[{"message":"Variable \"$input\" got invalid value
[\"ERROR_WRONG_CAPTCHA_ID\"] at \"input.recaptchaResponse\"; Expected type
String. String cannot represent a non string value:
[\"ERROR_WRONG_CAPTCHA_ID\"]","locations":[{"line":1,"column":20}],"extensions":{"errorCode":"INTERNAL_SERVER_ERROR"}}]}

**Comment by EdwardOak** [comment]
root@postbot2:/home/laXXXXXXXak/kijiji-bot/examples/repost_ads# python3
__main__.py
Logging in with account credentials...
['OK', '65380799054']
Captcha Currently Being Solved ...
['ERROR_WRONG_CAPTCHA_ID']
Traceback (most recent call last):
  File "__main__.py", line 25, in <module>
    config['cookies'] = bot.login(_USERNAME, _PASSWORD)
  File "/usr/local/lib/python3.6/dist-packages/kijiji_bot/__init__.py",
line 31, in login
    super().login(username, password)
  File "/usr/local/lib/python3.6/dist-packages/kijiji_bot/kijiji_api.py",
line 122, in login
    raise KijijiApiException("Could not log in.", resp.text)
kijiji_bot.kijiji_api.KijijiApiException: Could not log in.

**Comment by EdwardOak** [comment]
Please answer back, I need that thing to work :(

**Comment by EdwardOak** [comment]
All fixed!

I had removed your indexes ([1]) in Captcha.py for testing purposes as it
crashed with them when the captcha service balance was ZERO.
Removing indexes allowed me to progress and see the ZERO BALANCE error from
2Captcha, but prevented it from working when balance was present.

Thank you, sir.

Reach out to me if you need me for anything SMS/VoIP related and marketing
campaigns.

oakandrocks@gmail.com | My name's Edward, btw.

**Comment by Safeer-Abbas** [comment]
Hey,

I actually learned how to bypass SSL Pinning recently. Planning to make the kijiji bot using the sms api. That way it would be fast and there will be no re-captcha. Ill sniff out the api tomorrow.

**Comment by EdwardOak** [comment]
Do you discord?


⁣Sent from BlueMail ​

**Comment by Safeer-Abbas** [comment]
Wisdom#1984
